azure-ts-dynamicresource/index.ts

```diff
@@ -0,0 +1,96 @@
+
+// Create an Azure Resource Group
+const resourceGroup = new azure.core.ResourceGroup("resourceGroup", {
+    location: location,
+});
+
+// Create an Azure resource (Storage Account) so we can point the CDN endpoint to it.
+const storageAccount = new azure.storage.Account("storageAccount", {
+    resourceGroupName: resourceGroup.name,
+    accountTier: "Standard",
+    accountReplicationType: "LRS",
+});
+
+/**
+ * Create a Blob container in the storage account,
+ * to store any static files. The CDN endpoint will be pointed at the
+ * endpoint for this blob container.
+ */
+const blobContainer = new azure.storage.Container("blob-container", {
+    resourceGroupName: resourceGroup.name,
+    storageAccountName: storageAccount.name,
+    // Make each "blob" in the container publicly accessible.
+    // DO NOT set this property if you are going to store sensitive files!
+    containerAccessType: "blob",
+});
+
+const cdnProfile = new azure.cdn.Profile("cdn-profile", {
+    resourceGroupName: resourceGroup.name,
+    // Choose an appropriate SKU to use.
+    // https://docs.microsoft.com/en-us/azure/cdn/cdn-features
+    sku: "Standard_Akamai",
+});
+
+const cdnEndpoint = new azure.cdn.Endpoint("my-cdn-endpoint", {
+    name: "my-cdn-endpoint",
+    resourceGroupName: resourceGroup.name,
+    profileName: cdnProfile.name,
+    isHttpsAllowed: true,
+    isHttpAllowed: false,
+    isCompressionEnabled: true,
+    originHostHeader: storageAccount.primaryBlobHost,
+    contentTypesToCompresses: [
+        "text/plain",
+        "text/html",
+        "text/css",
+        "text/javascript",
+        "application/x-javascript",
+        "application/javascript",
+        "application/json",
+        "application/xml",
+        "image/png",
+        "image/jpeg",
+    ],
+    origins: [
+        {
+            name: "cdn-origin",
+            hostName: storageAccount.primaryBlobHost,
+            httpsPort: 443,
+        }
+    ]
+});
+
+export const cdnEndpointUrl = pulumi.interpolate `https://${cdnEndpoint.hostName}`;
+
+pulumi.all([resourceGroup.name, cdnProfile.name, cdnEndpoint.name])
+    .apply(([resourceGroupName, cdnProfileName, cdnEndpointName]) => {
+        cdnCustomDomainResource = new CDNCustomDomainResource("cdnCustomDomain", {
+            resourceGroupName: resourceGroupName,
+            // Ensure that there is a CNAME record for mycompany.com
+            // pointing to my-cdn-endpoint.azureedge.net.
+            // You would do that in your domain registrar's portal.
+            customDomainHostName: "mycompany.com",
+            profileName: cdnProfileName,
+            endpointName: cdnEndpointName,
+            // This will enable HTTPS through Azure's one-click
+            // automated certificate deployment.
+            // The certificate is fully managed by Azure from provisioning
+            // to automatic renewal at no additional cost to you.
+            httpsEnabled: true,
+        }, { parent: cdnEndpoint });
```

Comment:
i'm a little confused why any of this would be needed.  can't you just write:

```ts
cdnCustomDomainResource = new ...(... {
    resourceGroupName: resourceGroup.name,
    // etc.?
}
```

I don't see why you need the .all or .apply.

Reply:
The reason is it isn't clear from the docs about the dynamic provider if one can pass `Input<T>` props to the dynamic resource and still get strong typing in the provider's functions for the inputs.